Automattic/Gravatar-SDK-iOS
# Adding api key header to v3 profile request

Closes #256 

### Description

Adds a Configuration class to hold the API key globally in memory to be used in requests.
Using this api key in v3 Profile request.

For the demo app, I've added a `Secrets.swift` file to the project, which is being ignored by source control.
This file will be automatically generated when running `make dev` or `make dev-demo`.
It can also be generated specifically with `make secrets`.

I tried to make this happen in the build phase of the project, but there are several problems, like the new file is not immediately added to the current build process, which makes the build fail. And fastlane don't use xcode to build the demo apps.

After this file is generated, it won't be overridden, and we can keep our personal api-key in this generated file without worrying about committing it to source control.

### Testing Steps

- run `make dev-demo`
- Check that `Secrets.swift` has been generated.
- Add your api key to the new `Secrets.swift` file.
- Test the "Fetch Profile" demo screen.
- 🎉
